tests/server/test_plugins.py

```diff
@@ -43,79 +137,293 @@ def test_example_config(self):
+                                       params=params)
+        time.sleep(2.5)
+        assert not plugin.current_run
+        plugin.run()
+        assert plugin.current_run
+        assert type(plugin.current_run).__name__ == 'PluginWorker'
+
+    def test_run_plugin_with_interval_should_schedule_next_run(self):
+        params = dict(self.test_params_default, **{'Interval': 2})
+        plugin = server_plugins.Plugin(name=self.test_plugin_name,
+                                       params=params)
+        time.sleep(2.5)
+        plugin.run()
+        delta = (plugin.next_run - datetime.datetime.now())
+        assert round(delta.total_seconds(), 0) == plugin.params['Interval']
+
+    def test_plugin_should_not_be_run_without_interval_parameter(self):
+        plugin = server_plugins.Plugin(name=self.test_plugin_name,
+                                       params=self.test_params_default)
+        assert not plugin.next_run
+        plugin.forced = False
+        plugin.run()
+        assert not plugin.current_run
+
+    def test_interval_run_plugin_and_collect_new_result(self):
+        params = dict(self.test_params_default, **{'Interval': 2})
+        plugin = server_plugins.Plugin(name=self.test_plugin_name,
+                                       params=params)
+        expected_result = {'status': 'OK', 'forced': False}
+        time.sleep(2.5)
+        plugin.run()
+        time.sleep(0.5)
+
+        assert not plugin.get_last_result()
+        plugin.collect_new_result()
+        assert cmp_dict(plugin.get_last_result(), expected_result) == 2
+
+    def test_force_plugin_to_run_and_collect_new_result(self):
+        plugin = server_plugins.Plugin(name=self.test_plugin_name,
+                                       params=self.test_params_default)
+        expected_result = {'status': 'OK', 'forced': True}
+        plugin.forced = True
+        plugin.run()
+        time.sleep(0.5)
+        assert not plugin.get_last_result()
+        plugin.collect_new_result()
+        # plugin.force should switch to False after result is collected
+        assert not plugin.forced
+        assert cmp_dict(plugin.forced_result, expected_result) == 2
+
+    def test_history_of_collect_new_result(self):
+        params = dict(self.test_params_default, **{'History': 5})
+        plugin = server_plugins.Plugin(name=self.test_plugin_name,
+                                       params=params)
+        # expected_result = {'status': 'OK', 'forced': True}
+        runs = 7
+        for n in range(0, runs, 1):
+            plugin.forced = True
+            plugin.run()
+            time.sleep(0.5)
+            plugin.collect_new_result()
+
+        assert len(plugin.result) == params['History']
+        assert plugin.forced_result == plugin.result[-1]
+
 
-        worker = PluginWorker('test', queue, params)
-        worker.start()
-        result = queue.get()
-        worker.join()
-        self.assertNotEqual(result, None)
+def cmp_dict(a, b):
```

Comment:
What's the purpose of this function? There should probably be a comment explaining what it does...